apps/web/src/pages/Pools/ManageAccessPage/components/helpers.tsx

```diff
@@ -0,0 +1,131 @@
+import { IntlShape } from "react-intl";
+
+import { getLocalizedName } from "@gc-digital-talent/i18n";
+import { Link, Chip, Chips } from "@gc-digital-talent/ui";
+import { notEmpty } from "@gc-digital-talent/helpers";
+import {
+  Maybe,
+  ManageAccessPagePoolFragment as ManageAccessPagePoolFragmentType,
+} from "@gc-digital-talent/graphql";
+
+import RemovePoolMembershipDialog from "./RemovePoolMembershipDialog";
+import { ManageAccessPagePoolFragmentRoleType, PoolTeamMember } from "./types";
+
+function orderRoles(
+  roles: Array<ManageAccessPagePoolFragmentRoleType>,
+  intl: IntlShape,
+) {
+  return orderBy(roles, (role) => {
+    const value = getLocalizedName(role?.displayName, intl);
+
+    return value
+      ? value
+          .normalize("NFD")
+          .replace(/[\u0300-\u036f]/g, "")
+          .toLocaleLowerCase()
+      : value;
+  });
+}
+
+export const actionCell = (
+  user: PoolTeamMember,
+  pool: ManageAccessPagePoolFragmentType,
+  canRemoveRole: boolean,
+) => (
+  <div
+    data-h2-display="base(flex)"
+    data-h2-flex-wrap="base(wrap)"
+    data-h2-gap="base(x.25)"
+  >
+    <RemovePoolMembershipDialog
+      user={user}
+      pool={pool}
+      canRemoveRole={canRemoveRole}
+    />
+  </div>
+);
+
+export function emailLinkCell(
+  email: Maybe<string> | undefined,
+  intl: IntlShape,
+) {
+  if (email) {
+    return (
+      <Link color="black" external href={`mailto:${email}`}>
+        {email}
+      </Link>
+    );
+  }
+
+  return (
+    <span data-h2-font-style="base(italic)">
+      {intl.formatMessage({
+        defaultMessage: "No email provided",
+        id: "1JCjTP",
+        description: "Fallback for email value",
+      })}
+    </span>
+  );
+}
+
+export function roleCell(
+  roles: Maybe<Maybe<ManageAccessPagePoolFragmentRoleType>[]>,
+  intl: IntlShape,
+) {
+  const nonEmptyRoles = roles?.filter(notEmpty);
+  const roleChips = nonEmptyRoles
+    ? orderRoles(nonEmptyRoles, intl).map((role) => (
+        <Chip color="secondary" key={role?.id}>
+          {getLocalizedName(role?.displayName, intl)}
+        </Chip>
+      ))
+    : null;
+
+  return roleChips ? <Chips>{roleChips}</Chips> : null;
+}
+
+export function roleAccessor(
+  roles: Maybe<Maybe<ManageAccessPagePoolFragmentRoleType>[]>,
+  intl: IntlShape,
+) {
+  const nonEmptyRoles = roles?.filter(notEmpty);
+
+  return nonEmptyRoles
+    ? orderRoles(nonEmptyRoles, intl)
+        .map((role) => getLocalizedName(role?.displayName, intl))
+        .join(", ")
+    : "";
+}
+
+export const groupRoleAssignmentsByUser = (
+  assignments: ManageAccessPagePoolFragmentType["roleAssignments"],
+) => {
+  let users: Array<PoolTeamMember> = [];
+  const filteredAssignments = assignments?.filter((assignment) => {
+    return (
+      notEmpty(assignment.user) &&
+      notEmpty(assignment.role) &&
+      assignment.role.isTeamBased
+    );
+  });
+
+  filteredAssignments?.forEach((assignment) => {
+    const userIndex = users.findIndex(
```

Comment:
This is a minor, stylistic thing, but I think it would make this a bit more readable. This seems like a great place to use a [Map](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map)! Replace `let users = []` with `const users = new Map<string, PoolTeamMember>()`, with user ids as keys. That naturally represents what you want to do here with findIndex. Then, at the end of the function, return `users.values()` and you'll get the same array you want.

Reply:
Sure, I was just copying the function from below. But I can try to add that in https://github.com/GCTC-NTGC/gc-digital-talent/pull/11273/commits/d24ecb3074adc9e0c5c8690d0b5b24670b36f2bf

![image](https://github.com/user-attachments/assets/46009e93-50f0-4cd4-ab7d-2a936b4805e8)